File: senegal/src/main/java/com/sante/senegal/services/interfaces/ConsultationService.java
package com.sante.senegal.services.interfaces;

import com.sante.senegal.dto.ConsultationDto;
import com.sante.senegal.entities.Consultation;
import java.time.LocalDate;
import java.util.List;

public interface ConsultationService {
    ConsultationDto creerConsultation(Long rendezVousId, String symptomes, String diagnostic,
                                   String observations, String recommandations);
    ConsultationDto terminerConsultation(Long consultationId, Integer dureeReelle);
    ConsultationDto mettreAJourConsultation(Long consultationId, ConsultationDto dto);
    List<ConsultationDto> getHistoriqueConsultations(Long patientId);
    List<ConsultationDto> getConsultationsMedecin(Long medecinId, LocalDate dateDebut, LocalDate dateFin);
    ConsultationDto getConsultationById(Long id);
    List<ConsultationDto> getConsultationsPatient(Long patientId, LocalDate dateDebut, LocalDate dateFin);

}

File: medical-appointment-frontend/src/app/core/services/consultation.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface Consultation {
  id: number;
  rendezVousId: number;
  symptomes: string;
  diagnostic: string;
  observations: string;
  recommandations: string;
  dateConsultation: Date;
  dureeReelle: number;
  statut: string;
}

export interface CreateConsultationRequest {
  rendezVousId: number;
  symptomes: string;
  diagnostic: string;
  observations: string;
  recommandations: string;
}

export interface UpdateConsultationRequest {
  symptomes: string;
  diagnostic: string;
  observations: string;
  recommandations: string;
}

@Injectable({
  providedIn: 'root'
})
export class ConsultationService {
  private apiUrl = 'http://localhost:8081/api/v1/consultations';

  constructor(private http: HttpClient) {}

  creerConsultation(request: CreateConsultationRequest): Observable<Consultation> {
    return this.http.post<Consultation>(this.apiUrl, request);
  }

  mettreAJourConsultation(id: number, request: UpdateConsultationRequest): Observable<Consultation> {
    return this.http.put<Consultation>(`${this.apiUrl}/${id}`, request);
  }

  terminerConsultation(id: number, dureeReelle: number): Observable<Consultation> {
    const params = new HttpParams().set('dureeReelle', dureeReelle.toString());
    return this.http.put<Consultation>(`${this.apiUrl}/${id}/terminer`, null, { params });
  }

  getHistoriqueConsultations(patientId: number): Observable<Consultation[]> {
    return this.http.get<Consultation[]>(`${this.apiUrl}/patient/${patientId}/historique`);
  }

  getConsultationsMedecin(medecinId: number, dateDebut: string, dateFin: string): Observable<Consultation[]> {
    const params = new HttpParams()
      .set('dateDebut', dateDebut)
      .set('dateFin', dateFin);
    return this.http.get<Consultation[]>(`${this.apiUrl}/medecin/${medecinId}`, { params });
  }

  getConsultationById(id: number): Observable<Consultation> {
    return this.http.get<Consultation>(`${this.apiUrl}/${id}`);
  }
}